server/src/categories/categories.controller.ts

```diff
@@ -0,0 +1,95 @@
+import {
+    Controller,
+    Get,
+    Post,
+    Body,
+    Patch,
+    Param,
+    Delete,
+    UseGuards,
+    HttpException,
+    HttpStatus,
+  } from '@nestjs/common';
+  
+  import { CategoriesService } from './categories.service';
+  import { CreateCategoryDto } from './dto/create-category';
+  import { UpdateCategoryDto } from './dto/update-category';
+  // import { JwtAuthGuard } from '../auth/guards/jwt-auth.guard';
+  //authGaurd excluded because it is outside of the scope of this PR
+  import { DeleteResult } from 'typeorm';
+  import { Category } from './entities/category.entity';
+  
+  @Controller('categories')
+  export class CategoriesController {
+    constructor(private readonly CategoriesService: CategoriesService) {}
+  
+    @Post()
+    async create(
+      @Body() createCategoryDto: CreateCategoryDto,
+    ): Promise<Category | HttpException> {

```

Comment:
so many iterations on this haha
but yeah we actually don't need the `HttpException` in the return value
if we were retuning the `throw new ...` then we would
but we're throwing, which you can think of like a `break` for our whole server